examples/go-client/main.go

```diff
@@ -18,26 +18,76 @@ package main
 
 import (
 	"context"
+	"crypto/tls"
 	"fmt"
 	"log"
+
+	"github.com/gravitational/teleport/api/client"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/auth"
+
+	"github.com/pborman/uuid"
 )
 
 func main() {
+	ctx := context.Background()
 	log.Printf("Starting Teleport client...")
-	client, err := connectClient()
+
+	tlsConfig := &tls.Config{}
+	// Create valid tlsConfig here to load TLS credentials.
+
+	clt, err := auth.NewClient(client.Config{
+		// Addrs is the Auth Server address, must be local.
+		Addrs: []string{"localhost:3025"},
+		// Multiple credentials can be provided, and the first credentials to
+		// successfully authenticate an open connection to the client will be used.
+		Credentials: []client.Credentials{
+			client.LoadKeyPair(
+				"certs/access-admin.crt",
+				"certs/access-admin.key",
+				"certs/access-admin.cas",
+			),
+			client.LoadIdentityFile("certs/access-admin-identity"),
+			client.LoadTLS(tlsConfig),
```

Comment:
do we expect a typical user to only have on of  these?
if yes, let's keep the example with 1 type of credentials and leave a comment pointing to other loaders.

Reply:
The nice thing about leaving multiple methods here is that the user can choose their method, generate the certs, and run the program without changing this code. If I remove the additional methods, it gives them more work to add their preferred method back in. But I'll take out the TLS one since it's confusing regardless, and if you feel strongly about it, I'll remove the `LoadKeyPair`.